language/diem-vm/src/script_to_script_function.rs

```diff
@@ -0,0 +1,531 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Module containing remapping information between script hashes and their remapped script
+//! function
+
+use move_core_types::{identifier::IdentStr, language_storage::ModuleId};
+
+macro_rules! ADD_CURRENCY_TO_ACCOUNT_BYTES {
```

Comment:
Why macros instead of the bytes from LegacyStdLibScript?

Reply:
I'm not really convinced the VM should depend on that

Reply:
But will defer to @tzakian

Reply:
I hope we can eventually delete LegacyStdLibScript and have this file be the only record of the bad old ways.

Reply:
Hopefully! Although I think LegacyStdlibScripts might be sticking around for a decent while :( as the json-rpc interface still needs them to name transaction scripts...

Reply:
After sleeping on it I have reached a few points:
- I would be fine with linking against the legacy scripts, but yes eventually we would have to cut that cord (even if it is many years from now)
- I am fine with directly writing out the binaries as I have done already because they are set in stone and unmoving. In the same way, I am fine with writing out "AccountAdministrationScripts" as opposed to parsing a Move file and finding the name of the module and it's script functions 

So I could go either way, but I would prefer to just have them written out as I've done here

Reply:
I'm fine with directly writing these bytes down. 

One thing that we definitely need here is a test to make sure that these don't drift from the legacy script bytes. To do this, could you add a dev dependency on the compiled_stdlib and check that the script bytes in these macros match up with the script bytes in the `LegacyStdlibScript` bytes. That way if there's any drift we'll catch in the tests/CI.

Reply:
One quick question/nit: why macros and not `const`s here? Does Rust complain about `Size`?

Reply:
I couldn't get the pattern matching to work. I only got it to work with `_ if script_bytes == CONST` 
I figure the slice pattern will be faster than the actual equality call, but who knows